packages/slate/src/commands/at-range.js

```diff
@@ -807,36 +807,86 @@ Commands.insertFragmentAtRange = (editor, range, fragment) => {
   })
 }
 
-const findInsertionNode = (fragment, document, startKey) => {
-  const hasSingleNode = object => {
-    if (!object || object.object === 'text') return
-    return object.nodes.size === 1
-  }
-
-  const firstNode = object => object && object.nodes.first()
-  let node = fragment
+/**
+ * Get the deepest single child block inside `fragment` whose reversed block
+ * ancestors match the reversed block ancestors of the `document` starting at
+ * the `documentKey`.
+ *
+ * @param {Document} document
+ * @param {string} documentKey
+ * @param {Document} fragment
+ * @return {Node}
+ */
 
-  if (hasSingleNode(fragment)) {
-    let fragmentInner = firstNode(fragment)
+const findInsertionNode = (fragment, document, documentKey) => {
+  // Find the deepest block in a doc with no siblings.
+  const deepestSingleBlock = doc => {
```

Comment:
Note that this will return `doc` even if doc has multiple child nodes. Is that intended?